addons/mrp/migrations/13.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 Payam Yasaie <https://www.tashilgostar.com>
+# Copyright 2020 Andrii Skrypka <andrijskrypa@ukr.net>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_model_renames = [
+    ('mrp.subproduct', 'mrp.bom.byproduct'),
+]
+
+_table_renames = [
+    ('mrp_bom_line_product_attribute_value_rel', 'mrp_bom_line_product_template_attribute_value_rel'),
+]
+
+_mrp_subproduct_table_renames = [
+    ('mrp_subproduct', 'mrp_bom_byproduct'),
+]
+
+_field_renames = [
+    ('mrp.workorder', 'mrp_workorder', 'final_lot_id', 'finished_lot_id'),
+    ('mrp.bom', 'mrp_bom', 'sub_products', 'byproduct_ids'),
+    ('mrp.bom.line', 'mrp_bom_line', 'attribute_value_ids', 'bom_product_template_attribute_value_ids'),
+    ('mrp.production', 'mrp_production', 'propagate', 'propagate_cancel'),
+    ('stock.move', 'stock_move', 'subproduct_id', 'byproduct_id'),
+]
+
+_column_renames = {
+    'stock_move_line': [
+        ('lot_produced_id', None),
+    ],
+}
+
+_column_copies = {
+    'mrp_production': [
+        ('availability', 'reservation_state', None),
+    ],
+    'mrp_bom_line_product_template_attribute_value_rel': [
+        ('product_attribute_value_id', 'product_template_attribute_value_id', None),
+    ],
+}
+
+_xmlid_renames = [
+    # ir.model.access
+    ('mrp_byproduct.access_mrp_subproduct_manager', 'mrp.access_mrp_bom_byproduct_manager'),
+    ('mrp_byproduct.access_mrp_subproduct_user', 'mrp.access_mrp_bom_byproduct_user'),
+]
+
+
+def fill_bom_product_template_attribute_value(env):
+    """ Convert product.attribute.value ids to product.template.attribute.value ids"""
+    openupgrade.logged_query(env.cr, """
+    UPDATE mrp_bom_line_product_template_attribute_value_rel mbl_ptav_rel
+    SET product_template_attribute_value_id = (
+        SELECT id
+        FROM product_template_attribute_value ptav
+        WHERE mb.product_tmpl_id = ptav.product_tmpl_id
+            AND mbl_ptav_rel.product_attribute_value_id = ptav.product_attribute_value_id
+        LIMIT 1
+    )
+    FROM mrp_bom_line mbl
+        JOIN mrp_bom mb ON mbl.bom_id = mb.id
+    WHERE mbl.id = mbl_ptav_rel.mrp_bom_line_id
+    """)
+
+
+def mapped_reservation_state(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE mrp_production
+        SET reservation_state = CASE
+          WHEN reservation_state = 'partially_available' THEN 'assigned'
+          WHEN reservation_state = 'none' THEN NULL
+          END
+        WHERE reservation_state in ('partially_available', 'none')

```

Comment:
please change this line to `WHERE reservation_state in ('partially_available', 'confirmed')`.
Without this fix every mrp_production record in this state gives error when trying to start/stop working on its workorders.

Reply:
`reservation_state` was `availability` in v12, `availability` did not have value `confirmed`

Reply:
ok right, so the error is in `mrp_workorder.production_availability`, the field is defined as 
```
    production_availability = fields.Selection(
        'Stock Availability', readonly=True,
        related='production_id.reservation_state', store=True,
        help='Technical: used in views and domains only.')
```

In this case we should add another migration query, because it contains wrong values:

```
    openupgrade.logged_query(
        env.cr, """
        UPDATE mrp_workorder
        SET production_availability = CASE
          WHEN production_availability = 'partially_available' THEN 'assigned'
          WHEN production_availability = 'none' THEN NULL
          END
        WHERE production_availability in ('partially_available', 'none')
        """
    )
```

Reply:
Try with changes from https://github.com/kos94ok-3D/OpenUpgrade/commit/dc1f2321d42ecf7be7f7340a8e757c65f818b8e0

Reply:
Thanks, I'm trying right now, I'll let you know if it works

Reply:
@kos94ok-3D it keeps breaking when I try to start a workorder. Launching this query solves the problem:

```sql
UPDATE mrp_workorder
SET production_availability = CASE
  WHEN production_availability = 'partially_available' THEN 'assigned'
  WHEN production_availability = 'none' THEN NULL
  END
WHERE production_availability in ('partially_available', 'none')
```

Reply:
Add in the end of function
<pre>
    # production_availability of mrp.workorder because it related and store=True
    openupgrade.logged_query(
        env.cr, """
        UPDATE mrp_workorder wo
        SET reservation_state = mo.reservation_state
        FROM mrp_production mo
        WHERE wo.production_id = mo.id
        """
    )
</pre>

Reply:
Yes, with this I think that all works perfectly